### PR TITLE
Replace try/finally with try-with-resources.

### PR DESCRIPTION
**Pull request by jmark99**
Implement the try-with-resources construct available in Java 7+ where applicable.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java [hunk 1]
```diff
@@ -85,16 +85,10 @@ String process(String value) {
             justification = "app is run in same security context as user providing the filename")
         @Override
         String process(String value) {
-          Scanner scanner = null;
-          try {
-            scanner = new Scanner(new File(value), UTF_8);
+          try (Scanner scanner = new Scanner(new File(value), UTF_8)) {
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java]: This is a good opportunity to use `var`:

```suggestion
          try (var scanner = new Scanner(new File(value), UTF_8)) {
```

File: core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java [hunk 2]
```diff
@@ -188,17 +187,12 @@ public boolean testClassLoad(final String className, final String asTypeName)
 
   @Override
   public void ping(String tserver) throws AccumuloException {
-    TTransport transport = null;
-    try {
-      transport = createTransport(AddressUtil.parseAddress(tserver, false), context);
-      var client = createClient(new TabletClientService.Client.Factory(), transport);
+    try (
+        TTransport transport = createTransport(AddressUtil.parseAddress(tserver, false), context)) {
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java]:
```suggestion
    try (var transport = createTransport(AddressUtil.parseAddress(tserver, false), context)) {
```

File: core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/PrintInfo.java [hunk 3]
```diff
@@ -39,10 +39,8 @@
   public static void printMetaBlockInfo(SiteConfiguration siteConfig, Configuration conf,
       FileSystem fs, Path path) throws IOException {
     FSDataInputStream fsin = fs.open(path);
-    BCFile.Reader bcfr = null;
-    try {
-      bcfr = new BCFile.Reader(fsin, fs.getFileStatus(path).getLen(), conf,
-          CryptoServiceFactory.newInstance(siteConfig, ClassloaderType.ACCUMULO));
+    try (BCFile.Reader bcfr = new BCFile.Reader(fsin, fs.getFileStatus(path).getLen(), conf,
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/PrintInfo.java]:
```suggestion
    try (var bcfr = new BCFile.Reader(fsin, fs.getFileStatus(path).getLen(), conf,
```

Comment (jmark99) [reply core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/PrintInfo.java]: @ctubbsii thanks for the info. I'm not that familiar with the use of var at the moment so I will read up on it a bit and then see if I can apply it where appropriate. You can take another look once I push another update.

Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/PrintInfo.java]: That's fine. We can merge this without those suggestions. They aren't necessary.

**Comment by ctubbsii** [comment]
@jmark99 I went ahead and merged what you had, since it was fine. Any `var` improvements can be done as follow-on... if you even want to (it's not really necessary).

**Pull request merged**